--- sindresorhus---pure/pure.zsh	2020-05-03 23:43:20.780465483 -0400
+++ sindresorhus---pure/pure.zsh-patch	2020-05-03 23:47:22.214491671 -0400
@@ -131,12 +131,6 @@
 	# Initialize the preprompt array.
 	local -a preprompt_parts
 
-	# Username and machine, if applicable.
-	[[ -n $prompt_pure_state[username] ]] && preprompt_parts+=($prompt_pure_state[username])
-
-	# Set the path.
-	preprompt_parts+=('%F{${prompt_pure_colors[path]}}%~%f')
-
 	# Git branch and dirty status info.
 	typeset -gA prompt_pure_vcs_info
 	if [[ -n $prompt_pure_vcs_info[branch] ]]; then
